src/ipc/ipc4/handler.c

```diff
@@ -1003,14 +1003,48 @@ void ipc_boot_complete_msg(struct ipc_cmd_hdr *header, uint32_t data)
 }
 
 #ifdef CONFIG_LOG_BACKEND_ADSP_MTRACE
+
+static bool is_notification_queued(void)
+{
+	struct ipc *ipc = ipc_get();
+	struct list_item *slist;
+	struct ipc_msg *msg;
+	k_spinlock_key_t key;
+	bool queued = false;
+
+	key = k_spin_lock(&ipc->lock);
+	if (list_is_empty(&ipc->msg_list))
+		goto out;
+
+	list_for_item(slist, &ipc->msg_list) {
+		msg = container_of(slist, struct ipc_msg, list);
+
+		if (msg == &msg_notify) {
+			queued = true;
+			break;
+		}
+	}
+
+out:
+	k_spin_unlock(&ipc->lock, key);
+
+	return queued;
+}
+
 void ipc_send_buffer_status_notify(void)
 {
+	/* a single msg_notify object is used */
+	if (is_notification_queued())
```

Comment:
At the moment this should be OK as the only notification provider is mtrace, but in future if we have other notification sources then either each will have it's own static struct or we need to dynamically allocate notification messages..